src/h5web/visualizer/utils.ts

```diff
@@ -2,19 +2,30 @@ import type { HDF5Entity, HDF5Metadata } from '../providers/models';
 import { VIS_DEFS, Vis } from '../visualizations';
 
 export function getSupportedVis(
-  entity: HDF5Entity,
+  entity: HDF5Entity | undefined,
   metadata: HDF5Metadata
-): Vis[] {
-  const supported = Object.entries(VIS_DEFS).reduce<Vis[]>(
-    (arr, visDefEntry) => {
-      const [vis, { supportsEntity }] = visDefEntry;
-      return supportsEntity(entity, metadata) ? [...arr, vis as Vis] : arr;
-    },
-    []
-  );
+): { supportedVis: Vis[]; error?: Error } {
+  if (!entity) {
+    return { supportedVis: [] };
+  }
 
-  // Remove Raw vis if any other vis is supported
-  return supported.length > 1
-    ? supported.filter((vis) => vis !== Vis.Raw)
-    : supported;
+  try {
+    const supported = Object.entries(VIS_DEFS).reduce<Vis[]>(
+      (arr, visDefEntry) => {
+        const [vis, { supportsEntity }] = visDefEntry;
+        return supportsEntity(entity, metadata) ? [...arr, vis as Vis] : arr;
+      },
+      []
+    );
+
+    // Remove Raw vis if any other vis is supported
+    const supportedVis =
+      supported.length > 1
+        ? supported.filter((vis) => vis !== Vis.Raw)
+        : supported;
+
+    return { supportedVis };
+  } catch (error) {
+    return { supportedVis: [], error };
+  }
```

Comment:
The `try` wraps the `reduce`, so the function returns on the first error thrown by any of the `supportsEntity` functions.

Reply:
This was my first idea, but it breaks the rule of hooks.

Reply:
For which hook ?

Reply:
`const [activeVis, setActiveVis] = useState<Vis>();`

Reply:
As discussed, I can move the hook up but the `try/catch` has to wrap the rest of the body of the `Visualizer` function, so it's pretty ugly.